select.js

```diff
@@ -1,14 +1,39 @@
-define(function() {
-    var init = function() {
-        $('body').on('change', '.c-select', function() {
-            var $container = $(this);
-            var value = $container.find('option:selected').text();
-
-            $container.find('.c-select__value').text(value);
-        });
-    };
+define(['$'], function($) {
+    //
+    // Component constructor
+    //
+    var Select = function Select($el, options) {
+        // Component instance properties. The `$` hash (by convention) caches
+        // the root node and any sub-components.
+        this.value = '';
+        this.dom = {
+            $root: $el,
+            $value: $el.find('.c-select__value')
+        };
+
+        // Event handlers should use $.proxy to preserve access to the instance
+        // as `this` within the handler.
+        this.dom.$root.on('change', $.proxy(this.update, this));
```

Comment:
Also, `this` sucks in general, so if there’s some way to avoid plastering it all over component ui while maintaining the general notion of class + instance, that would be nice.